cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManagerTest.java

```diff
@@ -0,0 +1,337 @@
+        // Mock the Admin client
+        Admin mockAdminClient = mock(Admin.class);
+
+        // Mock describing the current metadata version
+        mockDescribeQuotasResultPresent(mockAdminClient);
+
+        QuotasPluginKafka quotasPluginKafka = new QuotasPluginKafkaBuilder()
+            .withProducerByteRate(DEFAULT_PRODUCER_BYTE_RATE)
+            .withConsumerByteRate(DEFAULT_CONSUMER_BYTE_RATE)
+            .withControllerMutationRate(DEFAULT_MUTATION_RATE)
+            .withRequestPercentage(DEFAULT_REQUEST_PERCENTAGE)
+            .build();
+
+        List<ClientQuotaAlteration.Op> desiredQuotas = DefaultKafkaQuotasManager.prepareQuotaConfigurationRequest(quotasPluginKafka);
+
+        DefaultKafkaQuotasManager.shouldAlterDefaultQuotasConfig(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClient, desiredQuotas, false)
+            .onComplete(context.succeeding(result -> {
+                context.verify(() -> assertFalse(result));
+                context.completeNow();
+            }));
+    }
+
+    /**
+     * Tests that the default Kafka quotas are altered in case of different configuration specified by user.
+     *
+     * @param context   Vertx test context
+     */
+    @Test
+    void testReconfigureDefaultQuotasSetInKafka(VertxTestContext context) {
+        long consumerByteRate = 1000;
+        long producerByteRate = 1000;
+        double mutationRate = 0.1;
+        int requestPercentage = 33;
+
+        Checkpoint checkpoint = context.checkpoint();
+
+        // Mock the Admin client
+        Admin mockAdminClient = mock(Admin.class);
+
+        // Mock describing the current metadata version
+        mockDescribeQuotasResultPresent(mockAdminClient);
+
+        // Mock the Admin Client Provider
+        AdminClientProvider mockAdminClientProvider = mockAdminClientProvider(mockAdminClient);
+
+        QuotasPluginKafka quotasPluginKafka = new QuotasPluginKafkaBuilder()
+            .withConsumerByteRate(consumerByteRate)
+            .withProducerByteRate(producerByteRate)
+            .withControllerMutationRate(mutationRate)
+            .withRequestPercentage(requestPercentage)
+            .build();
+
+        List<ClientQuotaAlteration.Op> expectedResult = DefaultKafkaQuotasManager.prepareQuotaConfigurationRequest(quotasPluginKafka);
+
+        // Scenario with configured QuotasPluginKafka and default user quota set in Kafka (all options) -> but with different values in both configurations
+        DefaultKafkaQuotasManager.reconcileDefaultUserQuotas(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClientProvider, DUMMY_IDENTITY.pemTrustSet(), DUMMY_IDENTITY.pemAuthIdentity(), quotasPluginKafka)
+            .onComplete(context.succeeding(s -> {
+                assertFalse(quotaAlternationToKafka.getValue().isEmpty());
+
+                List<ClientQuotaAlteration.Op> valuesSet = quotaAlternationToKafka.getValue().get(0).ops().stream().toList();
+                assertEquals(expectedResult, valuesSet);
```

Comment:
* Maybe you can use `assertTaht(...)`
* You should maybe check number of times how the Admin API method was called
* If you create the captor locally, the test will be much easier to read

Same applies to those below.